### DISCUSSION TITLE
Supporting for FFT

### DISCUSSION THREAD

**Discussion by pschatzmann** [pschatzmann/arduino-audio-tools#170] [Announcements]
Running FFT on audio data has been made very easy! 
Further [detail can be found in the Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/FFT)

**Comment by RWBRWB** [comment]
I'm using your A2DP library for Bluetooth Audio and I would like to display a Spectrum Analyzer to the 400x240 LCD I have to show the audio dancing as it plays.  

Something like this https://youtu.be/avjwpHCYAm0

Can you library do that using the I2S audio stream your A2DP library is creating?

**Comment by pschatzmann** [comment]
What this library is providing is an easy setup of some audio processing chain where you can e.g. send the output to a MultiOutput which forwards the data e.g. both to a I2SStream and an AudioRealFFT. Alternatively you can also use the data callback of the A2DP library to write the data to AudioRealFFT.

The result of the FFT will define the height of your bars. 

However, please note that this is an audio library and __not a LCD library__, therefore I don't provide any display functionality!

**Reply by RWBRWB** [reply]
Yea I can handle the LCD stuff, I just need ideally an example of your A2DP library that I am using now and linked together with the AudioRealFFT library and how to get 30-40 frequency variables that I can plot out on a LCD a few times a second if possible. 

Any example code of this you can link me to? 

Any help is appreciated. 

Your A2DP library is working great, use it daily.

**Comment by pschatzmann** [comment]
Here are some exaples: 
- [fft](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/streams-audiokit-fft/streams-audiokit-fft.ino)
- [multi output](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-sd_mp3-metadata/streams-sd_mp3-metadata.ino)

In any case I recommend that you have a look at the class documentation...
